crates/requirements-txt/src/lib.rs

```diff
@@ -296,31 +297,21 @@ impl Display for EditableRequirement {
 
 /// A [Requirement] with additional metadata from the requirements.txt, currently only hashes but in
 /// the future also editable an similar information
-#[derive(Debug, Deserialize, Clone, Eq, PartialEq, Hash, Serialize)]
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
 pub struct RequirementEntry {
     /// The actual PEP 508 requirement
     pub requirement: RequirementsTxtRequirement,
     /// Hashes of the downloadable packages
     pub hashes: Vec<String>,
 }
 
-impl Display for RequirementEntry {
-    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}", self.requirement)?;
-        for hash in &self.hashes {
-            write!(f, " --hash {hash}")?;
-        }
-        Ok(())
-    }
-}
-
 /// Parsed and flattened requirements.txt with requirements and constraints
 #[derive(Debug, Default, Clone, PartialEq, Eq)]
 pub struct RequirementsTxt {
     /// The actual requirements with the hashes.
     pub requirements: Vec<RequirementEntry>,
     /// Constraints included with `-c`.
-    pub constraints: Vec<Requirement>,
+    pub constraints: Vec<UvRequirement>,
```

Comment:
Shouldn't this be a `pep508::Requirement`?

Reply:
The problem is the entanglement with `RequirementsTxtRequirement`

Reply:
Can you say more? I still wouldn’t expect that this crate (which parses requirements files) needs to know about our requirement type (which doesn’t exist in requirements files). Couldn’t we convert back in the calling crate?

Reply:
(this comment is sorted above the one with the more detail, sorry) I tried not changing constraints but ran into two problems, one is that we parse the requirements and then push them as constraints (https://github.com/astral-sh/uv/blob/b18610fe78a754ef8b394e0c0eb8cd0bc2d51068/crates/requirements-txt/src/lib.rs#L478-L482). The other is that with `Constraints::apply` taking references to requirements and constraints that must be of the same type means that we either have to change the constraints to the new requirement too or we to rework (split?) a section of the requirements.txt parsing code/types.